neurovault/apps/statmaps/views.py

```diff
@@ -1202,16 +1202,18 @@ def get_initial_queryset(self):
 
 
 def download_collection(request, cid):
-    from django.http import StreamingHttpResponse
-
     # Files (local path) to put in the .zip
     collection = get_collection(cid, request)
-    filenames = [img.file.path for img in collection.basecollectionitem_set.all()]
+
+    filenames =[img.zip_file.path for img in collection.basecollectionitem_set.all()
+                if isinstance(img, NIDMResults)] + \
+               [img.file.path for img in collection.basecollectionitem_set.all()
+                if (isinstance(img, Image) and not isinstance(img, NIDMResultStatisticMap))]
```

Comment:
You can incorporate this into querysets using django polymorphic https://django-polymorphic.readthedocs.io/en/stable/advanced.html It should yield higher performance.